equinix/data_source_metal_spot_market_request.go

```diff
@@ -112,7 +113,7 @@ func dataSourceMetalSpotMarketRequestRead(ctx context.Context, d *schema.Resourc
 
 	d.SetId(id)
 
-	return equinix_schema.SetMap(d, map[string]interface{}{
+	err = equinix_schema.SetMap(d, map[string]interface{}{
```

Comment:
It's worth considering how SetMap (which depends on Hashicorp's go-multierror](https://github.com/hashicorp/go-multierror)) should be updated to work best with [diagnostics](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/diag#Diagnostics) 

I'm looking at the `Append` capabilities in https://developer.hashicorp.com/terraform/plugin/framework/diagnostics and wondering if we would adopt that in SetMap rather than using multierrors, or if this is something that we consider with Framework.

Reply:
I'm not sure that deeper integration with the diag package is worth the tradeoff of needing 2 implementations (or needing an additional shim to make it work for both)--the SDK and framwork diag packages have different interfaces, and `Append` is only defined for the framework implementation.

IMO the goal of the framework migration is to make it as easy as possible to move an SDK resource over to the framework, so having a single multierror-based SetMap that works for both SDK and framework is preferable until we complete the framework migration, and then once we're back on a single provider implementation we can optimize freely.

Reply:
I should have prefaced that this was more future thinking and not a direct review / blocker.